File: app/src/main/java/com/example/weizhenbin/wangebug/tools/permission/PermissionResultManager.java
package com.example.weizhenbin.wangebug.tools.permission;

/**
 * Created by weizhenbin on 2018/9/11.
 */

public class PermissionResultManager {

    private static PermissionResultManager manager;


    private IPermissionGrantResult iPermissionGrantResult;

    private IFloattingWindowPermissionGrantResult iFloattingWindowPermissionGrantResult;

    private PermissionResultManager(){

    }


    public void addFloattingWindowPermissionListener(IFloattingWindowPermissionGrantResult iFloattingWindowPermissionGrantResult){
        this.iFloattingWindowPermissionGrantResult=iFloattingWindowPermissionGrantResult;
    }
    public void addPermissionGrantListener(IPermissionGrantResult iPermissionGrantResult){
        this.iPermissionGrantResult=iPermissionGrantResult;
    }


    public IFloattingWindowPermissionGrantResult getiFloatingWindowPermissionGrantResult() {
        return iFloattingWindowPermissionGrantResult;
    }

    public IPermissionGrantResult getiPermissionGrantResult() {
        return iPermissionGrantResult;
    }

    public static PermissionResultManager getManager(){

        if (manager==null){
            synchronized (PermissionResultManager.class){
                if (manager==null){
                    manager=new PermissionResultManager();
                }
            }
        }
        return manager;
    }


}

File: app/src/main/java/com/example/weizhenbin/wangebug/tools/permission/PermissionResultManager.kt
package com.example.weizhenbin.wangebug.tools.permission

/**
 * Created by weizhenbin on 2018/9/11.
 */

class PermissionResultManager private constructor() {


    private var iPermissionGrantResult: IPermissionGrantResult? = null

    private var iFloatingWindowPermissionGrantResult: IFloattingWindowPermissionGrantResult? = null


    fun addFloatingWindowPermissionListener(iFloatingWindowPermissionGrantResult: IFloattingWindowPermissionGrantResult?) {
        this.iFloatingWindowPermissionGrantResult = iFloatingWindowPermissionGrantResult
    }

    fun addPermissionGrantListener(iPermissionGrantResult: IPermissionGrantResult?) {
        this.iPermissionGrantResult = iPermissionGrantResult
    }


    fun getiFloatingWindowPermissionGrantResult(): IFloattingWindowPermissionGrantResult? {
        return iFloatingWindowPermissionGrantResult
    }

    fun getiPermissionGrantResult(): IPermissionGrantResult? {
        return iPermissionGrantResult
    }




    private object SingletonHolder{
        val holder=PermissionResultManager()
    }



    companion object {

         fun getManager()=SingletonHolder.holder
    }


}
